gno.land/pkg/sdk/vm/handler.go

```diff
@@ -74,13 +76,14 @@ const (
 	QueryEval   = "qeval"
 	QueryFile   = "qfile"
 	QueryDoc    = "qdoc"
+	QueryPaths  = "qpaths"
 )
 
-func (vh vmHandler) Query(ctx sdk.Context, req abci.RequestQuery) abci.ResponseQuery {
-	var (
-		res  abci.ResponseQuery
-		path = secondPart(req.Path)
-	)
+func (vh vmHandler) Query(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
+	path := secondPart(req.Path)
+	if i := strings.IndexByte(path, '?'); i > 0 { // cut query
```

Comment:
```suggestion
	if i := strings.IndexByte(path, '?'); i >= 0 { // cut query
```

edge-case, but possible

Reply:
Doesn't seem addressed